File: src/main/java/com/ptah/entity/accounting/Account.java
package com.ptah.entity.accounting;

import com.ptah.common.BaseEntity;
import com.ptah.entity.contract.Participant;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Account extends BaseEntity {
    private String name;
    private BigDecimal balance;
    @OneToOne
    private Participant accountOwner;
}

File: src/main/kotlin/com/ptah/entity/accounting/Account.kt
package com.ptah.entity.accounting

import com.ptah.common.BaseEntity
import com.ptah.entity.contract.Participant
import jakarta.persistence.Entity
import jakarta.persistence.OneToOne
import lombok.*
import java.math.BigDecimal

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
class Account : BaseEntity() {
    private val name: String? = null
    private val balance: BigDecimal? = null

    @OneToOne
    private val accountOwner: Participant? = null
}
